aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -22,12 +22,42 @@ using namespace at::sparse;
 
 namespace {
 
-  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input) {
-    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true)) {
+  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input, bool promotes_integer_to_float = false) {
+    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true) && promotes_integer_to_float) {
```

Comment:
Consider using
```
template <bool promotes_integer_to_float = false>
SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input) ...
```

Reply:
@pearu Have done the changes as requested